js/modules/k6/http/response.go

```diff
@@ -137,6 +138,20 @@ func (res *HTTPResponse) Json() goja.Value {
 		default:
 			common.Throw(common.GetRuntime(res.ctx), errors.New("Invalid response type"))
 		}
+
+		if !gjson.ValidBytes(body) {
+			common.Throw(common.GetRuntime(res.ctx), errors.New("invalid json"))
+		}
```

Comment:
Hmm having this out of the `if hasSelector { /* ... */ }` check means that if users don't supply a selector to the `response.json()` call,  the response body would be validated twice - once by `gjson.ValidBytes()` and the second time by the stdlib's `json.Unmarshal()` - it will return an error for an invalid JSON response after all.

So I think this check should be inside of the `if hasSelector { /* ... */ }` scope. And as a slight optimization, you may also want to save the result of the first check in a new variable in the response, so that if users request multiple json values, the `body` is validated only once.